tests/testmongodb.nim

```diff
@@ -0,0 +1,219 @@
+import unittest
+
+import os, strutils, sequtils, times
+
+import norm / mongodb
+
+
+const
+  # for local testing, modify your /etc/hosts file to contain "mongodb_1"
+  # pointing to your local mongodb server
+  dbConnection = "mongodb://mongodb_1:27017"
+  dbName = "TestDb"
+  customDbName = "TestCustomDb"
+
+type
+  Copyright = object
+    legal_name: string
+    year: int
+    # year {.dbCol: "yr".}: int
+
+dbAddObject(Copyright)
+
+db(dbConnection, "", "", dbName):
+  type
+    User {.table: "users".} = object
+      email {.unique.}: string
+      ssn: Option[int]
+      birthDate: Time
+    Publisher {.table: "publishers".} = object
+      title {.unique.}: string
+    Book {.table: "books".} = object
+      title: string
+      authorEmail {.fk: User.email, onDelete: "CASCADE".}: string
+      publisherTitle {.fk: Publisher.title.}: string
+      ratings: seq[float]
+      copyright: Copyright

```

Comment:
I never tested fields of custom object types defined outside `db` scope. Does it work? If so, I think the test suite for SQLite and Postgres should have this test too.

Reply:
I would not work (or should not work) for SQLite as that database cannot store map columns. But expanding Postgres to have this makes sense. I think they are called composite types in Postgres.

We would need to add the "dbAddObject" macro as well.